File: script/crearProyecto.js
const fs = require('fs');
const path = require('path');

function crearEstructuraCarpetas() {
    const estructura = {
        'proyecto': {
            'app': ['config', 'controllers', 'core'],
            'public': [],
            'test': [],
            'views': []
        }
    };

    for (const carpetaPrincipal in estructura) {
        const carpetaPath = path.join(process.cwd(), carpetaPrincipal);

        if (!fs.existsSync(carpetaPath)) {
            fs.mkdirSync(carpetaPath);
            console.log(`Creada la carpeta: ${carpetaPrincipal}`);
        }

        for (const subcarpeta in estructura[carpetaPrincipal]) {
            const subcarpetaPath = path.join(carpetaPath, subcarpeta);

            if (!fs.existsSync(subcarpetaPath)) {
                fs.mkdirSync(subcarpetaPath);
                console.log(`Creada la subcarpeta: ${subcarpeta} en ${carpetaPrincipal}`);
            }

            for (const subcarpetaInterna of estructura[carpetaPrincipal][subcarpeta]) {
                const subcarpetaInternaPath = path.join(subcarpetaPath, subcarpetaInterna);

                if (!fs.existsSync(subcarpetaInternaPath)) {
                    fs.mkdirSync(subcarpetaInternaPath);
                    console.log(`Creada la subcarpeta: ${subcarpetaInterna} en ${subcarpeta}`);
                }
            }
        }
    }
}

crearEstructuraCarpetas();

File: script/crearProyecto.py
import os

def crear_estructura_carpetas():
    estructura = {
        'proyecto': {
            'app': ['config', 'controllers', 'core'],
            'public': [],
            'test': [],
            'views': []
        }
    }

    for carpeta_principal, subcarpetas in estructura.items():
        carpeta_path = os.path.join(os.getcwd(), carpeta_principal)

        if not os.path.exists(carpeta_path):
            os.makedirs(carpeta_path)
            print(f'Creada la carpeta: {carpeta_principal}')

        for subcarpeta, subcarpetas_internas in subcarpetas.items():
            subcarpeta_path = os.path.join(carpeta_path, subcarpeta)
            if not os.path.exists(subcarpeta_path):
                os.makedirs(subcarpeta_path)
                print(f'Creada la subcarpeta: {subcarpeta} en {carpeta_principal}')

            for subcarpeta_interna in subcarpetas_internas:
                subcarpeta_interna_path = os.path.join(subcarpeta_path, subcarpeta_interna)
                if not os.path.exists(subcarpeta_interna_path):
                    os.makedirs(subcarpeta_interna_path)
                    print(f'Creada la subcarpeta: {subcarpeta_interna} en {subcarpeta}')

if __name__ == "__main__":
    crear_estructura_carpetas()
